src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -11271,6 +11271,58 @@ class C { }";
 
             CleanupAllGeneratedFiles(srcFile.Path);
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        [WorkItem(37779, "https://github.com/dotnet/roslyn/issues/37779")]
+        public void CompilerWarnAsErrorDoesNotEmit(bool warnAsError)
```

Comment:
I'd also expect tests for the API, i.e. you call Emit and it should not actually emit to the stream.